onedal/_device_offload.py

```diff
@@ -194,17 +220,25 @@ def support_usm_ndarray(freefunc=False, queue_param=True):
 
     def decorator(func):
         def wrapper_impl(obj, *args, **kwargs):
-            usm_iface = _extract_usm_iface(*args, **kwargs)
+            usm_iface, input_array_api, input_array_api_device = _extract_array_attr(
+                *args, **kwargs
+            )
             data_queue, hostargs, hostkwargs = _get_host_inputs(*args, **kwargs)
             if queue_param and not (
                 "queue" in hostkwargs and hostkwargs["queue"] is not None
             ):
                 hostkwargs["queue"] = data_queue
             result = _run_on_device(func, obj, *hostargs, **hostkwargs)
-            if usm_iface is not None and hasattr(result, "__array_interface__"):
+            if usm_iface is not None and (
+                hasattr(result, "__array_interface__")
+                or isinstance(result, Iterable)
+                and hasattr(result[0], "__array_interface__")
```

Comment:
This only validates the first member of an iterable has an ```__array_interface``` as this code is to be very general, please make it safer with additional checks.

Reply:
Thank you for the comment. It gave the idea that actually the check here is redundant, since the check is already done on  the side of the `_copy_to_usm`. More impirtant check here is that the inputs have  usm_iface, that requires wrapping up outputs these format.

The same logic already done on the other wrapper `wrap_output_data`:  https://github.com/intel/scikit-learn-intelex/blob/4fd45689a9fc998c2f3228d4b066bf0e1aa04bf0/sklearnex/_device_offload.py#L92